### ISSUE TITLE
LastModified is not updated when source and destination are the same.

### ISSUE THREAD

**Issue by m15o** [adobe/S3Mock#811]
In Amazon S3, LastModified is updated when source and destination are the same.
You can confirm that the LastModified of list-obejects-v2 is updated by executing the following shell script:

```sh
BUCKET_NAME=<your_bucket_name>
aws s3api create-bucket --bucket "$BUCKET_NAME"
echo "hello world" | aws s3 cp - "s3://$BUCKET_NAME/test.txt"
aws s3api list-objects-v2 --bucket "$BUCKET_NAME"
sleep 1
aws s3api copy-object --bucket "$BUCKET_NAME" --key test.txt --copy-source "$BUCKET_NAME"/test.txt
aws s3api list-objects-v2 --bucket "$BUCKET_NAME"
```

Similarly, LastModified is **not** updated when source and destination are the same in s3mock.
You can confirm that the LastModified value of list-obejects-v2 is the same by running the following shell script:

```sh
docker run -d --name s3mock -p 9090:9090 -p 9191:9191 -t adobe/s3mock

BUCKET_NAME=s3mock-bucket
aws --endpoint-url http://localhost:9090 s3api create-bucket --bucket "$BUCKET_NAME"
echo "hello world" | aws --endpoint-url http://localhost:9090 s3 cp - "s3://$BUCKET_NAME/test.txt"
aws --endpoint-url http://localhost:9090 s3api list-objects-v2 --bucket "$BUCKET_NAME"
sleep 1
aws --endpoint-url http://localhost:9090 s3api copy-object --bucket "$BUCKET_NAME" --key test.txt --copy-source "$BUCKET_NAME"/test.txt
aws --endpoint-url http://localhost:9090 s3api list-objects-v2 --bucket "$BUCKET_NAME"
```

We want s3mock to update LastModified like Amazon S3 when source and destination are the same.

**Comment by afranken** [comment]
thanks for the report.

S3Mock indeed does not update any of the metadata when it is instructed to copy a key onto itself.
This use-case was completely broken until #468, but it seems that this detail slipped through.
I'll reproduce against S3 and then fix this.

**Comment by afranken** [comment]
@m15o 
Just released 2.6.2 that fixes this issue.
